service/src/main/java/com/epam/esm/service/validator/GiftCertificateValidator.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.esm.service.validator;
+
+import com.epam.esm.service.dto.GiftCertificateDto;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+
+@Component
+public class GiftCertificateValidator extends AbstractValidator<GiftCertificateDto> {
+  private static final int MIN_NAME_LENGTH = 3;
+  private static final int MAX_NAME_LENGTH = 50;
+  private static final double MIN_PRICE = 0.0d;
+
+  @Override
+  public boolean validate(GiftCertificateDto giftCertificateDto) {
+    eraseErrorMessages();
```

Comment:
The method is too long, if there are more checks, it will become less and less readable. You can use more specific private methods to avoid this

Reply:
Yes, thank you. It's a good idea to divide my long method. (fixed)

Reply:
Thank you for reviewing my code!